File: VB/ReportService/ChangeImageSizeModeActionInvoker.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraReports.UI
Imports CustomEditorSample

Namespace ReportService
	Public Class ChangeImageSizeModeActionInvoker
		Inherits DevExpress.XtraReports.Service.EditActionBaseInvoker
		Protected Overrides Sub ControlInvoke(ByVal control As XRControl, ByVal editAction As DevExpress.XtraReports.ServiceModel.DataContracts.EditActionBase)
			Dim action = TryCast(editAction, ChangeImageSizeModeAction)
			Dim pictureBox = TryCast(control, XRPictureBox)
			If action Is Nothing OrElse pictureBox Is Nothing Then
				Return
			End If
			pictureBox.Sizing = action.ImageSizeMode
		End Sub
	End Class
End Namespace

File: CS/ReportService/ChangeImageSizeModeActionInvoker.cs
using CustomEditorSample;
using DevExpress.DocumentServices.ServiceModel.DataContracts.Xpf.Designer;
using DevExpress.XtraReports.Service;
using DevExpress.XtraReports.UI;

namespace ReportService {
    public class ChangeImageSizeModeActionInvoker : EditActionBaseInvoker {
        protected override void ControlInvoke(XRControl control, EditActionBase editAction) {
            var action = editAction as ChangeImageSizeModeAction;
            var pictureBox = control as XRPictureBox;
            if(action == null || pictureBox == null)
                return;
            pictureBox.Sizing = action.ImageSizeMode;
        }
    }
}
